From: Eric Dorland <eric@debian.org>
Date: Sat, 16 Sep 2017 18:47:43 -0400
Subject: $SOURCE_DATE_EPOCH support for mdate-sh

Uses $SOURCE_DATE_EPOCH if it is set, to get reproducible behavior no
matter what file mtime says.
---
 lib/mdate-sh | 10 ++++++++++
 1 file changed, 10 insertions(+)

Index: automake-1.16/lib/mdate-sh
===================================================================
--- automake-1.16.orig/lib/mdate-sh	2018-07-18 20:02:46.619692936 -0400
+++ automake-1.16/lib/mdate-sh	2018-07-18 20:02:46.591693280 -0400
@@ -88,6 +88,16 @@
 
 save_arg1=$1
 
+# Try SOURCE_DATE_EPOCH
+if test -n "${SOURCE_DATE_EPOCH}"; then
+  if date -d "@${SOURCE_DATE_EPOCH}" '+%-d %B %Y'; then
+    exit 0
+  else
+    error 'Invalid $SOURCE_DATE_EPOCH'
+  fi
+fi
+  
+
 # Find out how to get the extended ls output of a file or directory.
 if ls -L /dev/null 1>/dev/null 2>&1; then
   ls_command='ls -L -l -d'
